tests/crashes/123141.rs

```diff
@@ -1,22 +1,23 @@
 //@ known-bug: #123141
-trait ConstChunksExactTrait<T> {
-    fn const_chunks_exact<const N: usize>(&self) -> ConstChunksExact<'_, T, { N }>;
+
+trait Trait {
```

Comment:
Had to rewrite this test as this PR masked the ICE due to the fact that `const N: usize` now introduces a bound that is ambiguous for an inference variable.